src/main/java/org/folio/services/finance/transaction/PendingPaymentWorkflowService.java

```diff
@@ -96,6 +96,10 @@ private Future<Void> createPendingPayments(List<InvoiceWorkflowDataHolder> holde
             Future<Void> future = baseTransactionService.createTransaction(pendingPayment, requestContext)
               .recover(t -> {
                 logger.error("Failed to create pending payment with id {}", pendingPayment.getId(), t);
+                if (t instanceof HttpException) {
+                  HttpException exception = (HttpException) t;
+                  throw new HttpException(exception.getCode(), exception.getErrors());
```

Comment:
As for exception is the expected result and you use `.recover()`, please change this code to `return Future.failedFuture(new HttpException(...))`. As another option you could use `.otherwise` method which is more applicable for the blocking code